database/buildSchema/18_application.sql

```diff
@@ -55,3 +55,49 @@ CREATE TRIGGER outcome_trigger
     WHEN (OLD.outcome = 'PENDING' AND NEW.outcome <> 'PENDING')
     EXECUTE FUNCTION public.outcome_changed ();
 
+--FUNCTION to revert application status/active when OUTCOME is changed back to PENDING
+CREATE OR REPLACE FUNCTION public.outcome_reverted ()
+    RETURNS TRIGGER
+    AS $application_event$
+BEGIN
+    UPDATE
+        public.application
+    SET
+        is_active = TRUE
+    WHERE
+        id = NEW.id;
+    INSERT INTO public.application_status_history (application_stage_history_id, status)
+        VALUES ((
+                SELECT
+                    id
+                FROM
+                    application_stage_history
+                WHERE
+                    application_id = NEW.id
+                    AND is_current = TRUE),
+                (
+                    SELECT
+                        status
+                    FROM
+                        application_status_history
+                    WHERE
+                        time_created = (
+                            SELECT
+                                MAX(time_created)
+                            FROM
+                                application_status_history
+                            WHERE
+                                is_current = FALSE
+                                AND application_id = NEW.id)));
```

Comment:
doesn't look like there is application_id on `application_status_history`. Probably need to use `application_stage_status_all`

Reply:
Yeah, there is -- it's an auto-computed column -- see `CREATE OR REPLACE FUNCTION public.application_status_history_application_id` in `21_application_status_history.sql`

<img width="836" alt="Screen Shot 2022-08-02 at 11 29 40 AM" src="https://user-images.githubusercontent.com/5456533/182262090-6eafb716-e73f-40b9-ba34-5134584aa91f.png">

Reply:
oops, thanks